drivers/flash/flash_simulator.c

```diff
@@ -397,10 +397,8 @@ static int flash_mock_init(struct device *dev)
 		return -EIO;
 	}
 
-	if (f_stat.st_size == 0) {
-		/* erase the memory unit by pulling all bits to one */
-		(void)memset(mock_flash, 0xff, FLASH_SIMULATOR_FLASH_SIZE);
-	}
+	/* erase the memory unit by pulling all bits to one */
```

Comment:
This breaks persistency of the flash device, the flash should only be set to all 0xFF if not already exists.

Reply:
OK I dropped that commit